### PR TITLE
debouncing to prevent performance degradation

### PR DESCRIPTION
**Pull request by akshayjai1**
https://davidwalsh.name/javascript-debounce-function
https://css-tricks.com/the-difference-between-throttling-and-debouncing/

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New Question
- [x] Revision of an existing question
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate) 

# Checklist:

- [ ] My prose follows the style guidelines of this project
- [ ] I have performed a self-review of my own prose

### CODE REVIEW

**Comment by roblarsen** [comment]
I like the topic. I'd prefer a more generic question about the general topic. Maybe comparing debounce to throttle and looking for example use cases.

**Comment by akshayjai1** [comment]
I watched this [video](https://www.youtube.com/watch?v=93g5JWAjtk0) , it explains a very simple implementation of debouncing
i.e.
(to clear the timer if it exists and create a new timer) on the event.

So that the timer's functionality will only execute, if there is no consecutive event, thus achieving debouncing.

with regards to your question, if we ask them difference between two, then the user must need to know these terms which is not quite obvious requirement and same is the case if we ask what is the use case of debouncing, thus imho, current way of asking about debouncing seems better.

However any comment is still welcome, to make me think more on topic.

**Comment by roblarsen** [comment]
Hi @akshayjai1 I would hope my comments are welcome as I help manage this project and I was giving you feedback on your pull request.  I don't think your question is ready to be pulled into the project, but I think it's a good topic, so I was helping you to get your question into a state where I'd be happy to accept your pull request. 

I think the point of an interview question is to test a person's knowledge and debounce/throttle are two closely related optimization topics. Asking a general question about both opens up an opportunity to talk about optimizations and potentially illustrate previous work (using one or the other technique) and/or knowledge of implementations of these techniques (e.g. lodash)

**Comment by akshayjai1** [comment]
Hi @roblarsen , highly appreciate your taking time to comment.

I would like to propose one more way to talk about same
i.e
Tell the interviewee that we want to improve the performance of a search functionality, by executing search logic only when user finished typing, and thus ask him, how would he approach to achieve this optimization ( i.e debouncing)

Similarly for throttling, we can ask him that we want to display mouse co-ords on screen, on mouse move, but only with interval of atleast t seconds ( example of throttling)

Thus it would be more of a logical and algorithmic question( at the same time, if use knows about debouncing or throttling, he will mention that it can also be done using libraries)

File: questions/javascript-questions.md [hunk 1]
```diff
@@ -64,5 +64,5 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are the benefits of using `spread syntax` and how is it different from `rest syntax`?
 * How can you share code between files?
 * Why you might want to create static class members?
-* How to implement performance improvement of a search functionality, by executing search logic only when user finished typing?
-* How to restrict a function to only execute with a gap of atleast t seconds?
+* How to implement performance improvement of a search functionality, by executing search logic only when user finished typing? (debouncing)
```
Comment (roblarsen) [inline questions/javascript-questions.md]: My previous feedback still stands. You have to make this question more generic. It's not a question about  search, it's a question about debouncing which can be applied to many different cases. coupling the question tightly to some specific use case might throw off someone unfairly. If you want to add a specific coding challenge, that's one thing. But testing for concepts has to be as **generic** as possible.  

I still think that a single, simple question that asks about both concepts is better, anyway. I would prefer that.

File: questions/javascript-questions.md [hunk 2]
```diff
@@ -64,5 +64,5 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
...
+* How to restrict a function to only execute with a gap of atleast t seconds? (throttling)
```
Comment (roblarsen) [inline questions/javascript-questions.md]: This doesn't actually test whether or not the user understands the concept. This is nuts and bolts, not conceptual. Someone could write this code and not know what throttling is or know that there are utilities available to provide this functionality. 

Please read through the questions in this file and pattern this section after those.

**Pull request closed without merging**